### ISSUE TITLE
JSON read error when processing repo query batches.

### ISSUE THREAD

**Issue by vlad43210** [uva-bi-sdad/GHOSS.jl#19]
**Describe the bug**
When executing find_repos(batch) from the [repos script](https://github.com/uva-bi-sdad/GHOST.jl/blob/main/scripts/03_repos.jl), I get the following error:

```
ERROR: ArgumentError: invalid JSON at byte position 0 while parsing type Any: UnexpectedEOF


Stacktrace:
 [1] invalid(::JSON3.Error, ::Base.CodeUnits{UInt8,String}, ::Int64, ::Type{T} where T) at /home/vscode/.julia/packages/JSON3/hC8pW/src/JSON3.jl:24
 [2] read(::String; jsonlines::Bool, kw::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at /home/vscode/.julia/packages/JSON3/hC8pW/src/read.jl:39
 [3] read(::String) at /home/vscode/.julia/packages/JSON3/hC8pW/src/read.jl:13
 [4] find_repos(::DataFrames.SubDataFrame{DataFrame,DataFrames.Index,Array{Int64,1}}) at /home/vscode/.julia/packages/GHOST/dIqhL/src/04_Repos.jl:45
 [5] top-level scope at ./REPL[47]:2
```

**Environment**
using the Ghost package via Pkg 

**Minimal Reproducible Example (MRE)**
run the [repos script](https://github.com/uva-bi-sdad/GHOST.jl/blob/main/scripts/03_repos.jl).

**Expected behavior**
If this is a bad data issue on Github's end, perhaps find_repos should be wrapped in a try / catch block? I am trying that for now.

**Comment by Nosferican** [comment]
Yes. That issues arises from GitHub sending bad data. Most of the errors are handled through try/catch (e.g., really uncommon errors like when GitHub refuses to send data due to a copyright takedown issue). This particular one was quite hard to work around due to the sporadic nature... I usually report the error to GitHub directly and their team takes a look at it and it's resolved promptly. I will take a look if I can do some more advanced handling like waiting and trying again a couple times before throwing the error. Some errors should not be handled too gracefully since if the API is having issues you want to exit the program and resume at a later time rather than keep spending computing resources in vain.
